### PR TITLE
Handle group's email from web UI

### PR DESCRIPTION
**Pull request by saraycp**
In order to improve the group subscriptions to different types of notifications, we start with providing a way to handle groups emails from the UI. Something that was only possible[ through API](https://api.opensuse.org/apidocs/index#/Groups/post_group__group_title_) so far.

Create:

Screenshot 2024-10-03 at 16-40-23 Create Group - Open Build Service

Update email:

Screenshot 2024-10-03 at 16-39-58 Edit Group - Open Build Service

Actions column (mail_to and edit emails):

Screenshot 2024-10-03 at 16-39-36 Manage Groups - Open Build Service

### CODE REVIEW

File: src/api/spec/controllers/webui/groups_controller_spec.rb [hunk 1]
```diff
@@ -89,4 +89,52 @@
...
+      put :update, params: { title: group.title, group: { email: email } }
+    end
+
+    context 'as a normal user' do
+      let(:login_as) { create(:user) }
+      let(:email) { 'new_email@example.com' }
+
+      it 'does not allow to update the group' do
+        expect(flash[:error]).to eq('Sorry, you are not authorized to update this group.')
+        expect(Group.where(email: email)).not_to exist
```
[the remark this answers is not in the document]
Comment (saraycp) [reply src/api/spec/controllers/webui/groups_controller_spec.rb]: Please have another look.

**Review by eduardoj — APPROVED** [review]

File: src/api/app/controllers/webui/groups_controller.rb [hunk 2]
```diff
@@ -36,6 +41,18 @@ def create
     redirect_back_or_to root_path, error: "Group can't be saved: #{group.errors.full_messages.to_sentence}"
   end
 
+  def update
+    @group = Group.find_by(title: params[:title])
```
Comment (danidoni) [inline src/api/app/controllers/webui/groups_controller.rb:45]: What happens if the group title is not found?

Comment (saraycp) [reply src/api/app/controllers/webui/groups_controller.rb:45]: In other actions we are also just assuming a correct  title comes, but OK, someone can type an inexistent title in the URL, so it doesn't harm to control it.  I'm going to do it in these actions.

Comment (saraycp) [reply src/api/app/controllers/webui/groups_controller.rb:51]: Please have another look

**Pull request merged**